server/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -383,11 +383,25 @@ public StorageSupport<Map<String, Object>> storageSupport() {
     }
 
     @Override
-    public long valueBytes(Map<String, Object> value) {
-        if (value == null) {
+    @SuppressWarnings("unchecked")
+    public long valueBytes(Map<String, Object> map) {
+        if (map == null) {
             return RamUsageEstimator.NUM_BYTES_OBJECT_HEADER;
         }
-        return RamUsageEstimator.sizeOfMap(value);
+        long bytes = 0;
```

Comment:
```suggestion
        long bytes = RamUsageEstimator.shallowSizeOf(map);
```

If I understand correctly, that's similar to `RamUsageEstimator.sizeOfMap` but with nesting, nested calls are triggered by innerTypes. I tried locally a copy of `RamUsageEstimator.sizeOfMap` with some dummy very high MAX_LEN and got the assertion:
```
Expected :2504L // RamUsageEstimator.sizeOfMap with increased depth limit
Actual   :2376L // This PR
```
 
Computation is diverged in 2 places:
1. We don't call `alignObjectSize(size)` at the end. Not sure it's important
2. We are missing shallow size of the Map on each level. That's object header + sizes if internal field of the Map. Even for the small object in the new test, we already got more than 100 bytes difference, for very deep objects, or for array with 100K nested maps it can be quite some difference..

Reply:
I re-used this code for GeoShape and Unchecked types and addressed this in the re-used method in https://github.com/crate/crate/pull/16706